### PR TITLE
Added setter functions + removed unnecessary computation of calc

### PR DESCRIPTION
**Pull request by cmastalli**
This PR proposes:
 - updates the example-robot-data version (and bipedal-ubound log) due to a fixed bug on this packages,
 - setter functions for the running / terminal models in shooting problem,
 - setter function for the differential action model in euler integrator,
 - `recalc` option for `calcDiff(data, x)` cases (action models),
 - `recalc` option for `calcDiff(xs, us)` in shooting problem, and
 - to avoid unnecessary extra computation of `calc` of each node

The latter point comes from the fact that we have already computed `calc` during the rollout. Thus, we don't call `calc` after the first iteration.

@proyan this is for you. I have tested the code against all log files + unittest code.

### CODE REVIEW

**Comment by cmastalli** [comment]
This PR is blocked by the release of example-robot-data v3.1.2.
I will restart the CI when it has been finished

**Comment by proyan** [comment]
Hey Carlos,
In this PR, you are doing a lot of things. It is hard to do a review.

The setter function is not what we need! What is needed for MPC is to be able to change one single parameter of the original problem, without having to redefine a new problem completely. These setters would be useless if you want to be efficient.

Simply add a non-const getter to the variables, as I initially asked for in #672 . Or make them unprotected.

**Comment by cmastalli** [comment]
@proyan 

To be able to set cost parameters in c++, the user needs to downcast various classes.
This is a working example in c++:
```cpp
  // Downcast an action model to integrated action model
  crocoddyl::IntegratedActionModelEuler* m0 =
      static_cast<crocoddyl::IntegratedActionModelEuler*>(ddp.get_problem()->get_terminalModel().get());
  // Downcast an euler integrated action model to specific differential action model
  crocoddyl::DifferentialActionModelFreeFwdDynamics* m1 =
      static_cast<crocoddyl::DifferentialActionModelFreeFwdDynamics*>(m0->get_differential().get());
  // Retrieve the cost-base model
  const boost::shared_ptr<crocoddyl::CostModelAbstract>& c =
      m1->get_costs()->get_costs().find("gripperPose")->second.cost;
  // Downcast the cost abstraction to specific cost
  crocoddyl::CostModelFramePlacement* mycost = static_cast<crocoddyl::CostModelFramePlacement*>(c.get());
```

To make the c++ easier to write, the solution is to define as much as possible the objects and properties in the abstract class. For instance, we will need to define a cost model in the ActionModelAbstract class.
Despite that this is possible, we need to agree that we want to do that.
Unfortunately, other cases are not possible. For instance, we cannot define differential in the action-base class.

This solution will reduce the among of needed downcast. I could work on it, but in another PR. I need to modify many places in the code.

Therefore, I will proceed to merge this PR. And continuing discussing with you about next improvements.

File: include/crocoddyl/core/solvers/ddp.hpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ class SolverDDP : public SolverAbstract {
   virtual double tryStep(const double& steplength = 1);
   virtual double stoppingCriteria();
   virtual const Eigen::Vector2d& expectedImprovement();
-  virtual double calc();
+  virtual double calcDiff();
```
Comment (proyan) [inline include/crocoddyl/core/solvers/ddp.hpp]: Hi Carlos,

Could we please avoid this type of arbitrary API changes? It breaks some of my code, and is a bit frustrating to find.

**Pull request merged**